Integration test flake TestRulerFederatedRules/__tenant_id___is_present_on_metrics_for_federated_rules_when_source_tenants_

**Describe the bug**

```
=== CONT  TestRulerFederatedRules
    ruler_test.go:728: 
        	Error Trace:	ruler_test.go:728
        	            				ruler_test.go:829
        	Error:      	"" should have 1 item(s), but has 0
        	Test:       	TestRulerFederatedRules
=== CONT  TestRulerFederatedRules/__tenant_id___is_present_on_metrics_for_federated_rules_when_source_tenants_==_owner
    testing.go:1169: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
21:15:42 Killing ingester
21:15:42 Killing distributor
21:15:42 Killing consul
21:15:42 Killing minio-9000
21:15:42 Killing querier
21:15:42 Killing ruler
21:15:43 Killed ingester
21:15:43 Killed querier
21:15:43 Killed minio-9000
21:15:43 Killed ruler
21:15:43 Killed consul
21:15:43 Killed distributor
--- FAIL: TestRulerFederatedRules (12.10s)
    --- PASS: TestRulerFederatedRules/separate_source_tenants_and_destination_tenant (1.36s)
    --- PASS: TestRulerFederatedRules/__tenant_id___is_added_on_all_metrics_for_federated_rules (1.87s)
    --- FAIL: TestRulerFederatedRules/__tenant_id___is_present_on_metrics_for_federated_rules_when_source_tenants_==_owner (1.43s)
FAIL
FAIL	github.com/grafana/mimir/integration	225.211s
?   	github.com/grafana/mimir/integration/ca	[no test files]
?   	github.com/grafana/mimir/integration/e2emimir	[no test files]
FAIL
```

**To Reproduce**

Can be observed as part of this CI run:

https://github.com/grafana/mimir/runs/4902093551?check_suite_focus=true

--- a/integration/ruler_test.go
+++ b/integration/ruler_test.go
@@ -801,6 +801,11 @@ func TestRulerFederatedRules(t *testing.T) {
 			c, err := e2emimir.NewClient(distributor.HTTPEndpoint(), querier.HTTPEndpoint(), "", ruler.HTTPEndpoint(), tc.ruleGroupOwner)
 			require.NoError(t, err)
 
+			// Obtain total series before rule evaluation
+			totalSeriesBeforeEval, err := ingester.SumMetrics([]string{"cortex_ingester_memory_series"})
+			require.NoError(t, err)
+
+			// Create federated rule group
 			namespace := "test_namespace"
 			ruleName := "federated_rule_name"
 			g := ruleGroupWithRule("x", ruleName, tc.ruleExpression)
@@ -819,10 +824,8 @@ func TestRulerFederatedRules(t *testing.T) {
 			require.Len(t, retrievedNamespace, 1)
 			require.ElementsMatch(t, retrievedNamespace[0].SourceTenants, tc.groupSourceTenants)
 
-			// Wait for at least one evaluation
-			ruleEvaluationsRightAfterPush, err := ruler.SumMetrics([]string{"cortex_prometheus_rule_evaluations_total"})
-			require.NoError(t, err)
-			require.NoError(t, ruler.WaitSumMetrics(e2e.Greater(ruleEvaluationsRightAfterPush[0]), "cortex_prometheus_rule_evaluations_total"))
+			// Wait until rule evaluation resulting series had been pushed
+			require.NoError(t, ingester.WaitSumMetrics(e2e.Greater(totalSeriesBeforeEval[0]), "cortex_ingester_memory_series"))
 
 			result, err := c.Query(ruleName, time.Now())
 			require.NoError(t, err)

